src/GafferScene/RenderManifest.cpp

```diff
@@ -56,10 +56,75 @@
 
 using namespace GafferScene;
 
+namespace
+{
+
+struct CacheKey
+{
+	CacheKey() {};
+	CacheKey( IECore::ConstStringDataPtr loadedFromCryptomatteJSONStringData, std::string loadedFromFilePath, std::filesystem::file_time_type loadedFromFilePathModTime )
+		: loadedFromCryptomatteJSONStringData( loadedFromCryptomatteJSONStringData ), loadedFromFilePath( loadedFromFilePath ), loadedFromFilePathModTime( loadedFromFilePathModTime )
+	{
+	}
+
+	bool operator == ( const CacheKey &other ) const
+	{
+		if( other.loadedFromFilePath != loadedFromFilePath || other.loadedFromFilePathModTime != loadedFromFilePathModTime )
+		{
+			return false;
+		}
+
+		if( bool( other.loadedFromCryptomatteJSONStringData ) != bool( loadedFromCryptomatteJSONStringData ) )
+		{
+			return false;
+		}
+
+		if( ( !other.loadedFromCryptomatteJSONStringData ) && ( !loadedFromCryptomatteJSONStringData ) )
+		{
+			return true;
+		}
+
+		return other.loadedFromCryptomatteJSONStringData->isEqualTo( loadedFromCryptomatteJSONStringData.get() );
+	}
+
+	IECore::ConstStringDataPtr loadedFromCryptomatteJSONStringData;
+	std::string loadedFromFilePath;
+	std::filesystem::file_time_type loadedFromFilePathModTime;
+};
+
+
+struct CacheKeyHasher
+{
+	std::size_t operator()( const CacheKey &key ) const
+	{
+		IECore::MurmurHash h;
+		// Don't want to hash the actual contents of what might be an extremely large string.
+		// Just hashing the address will handle the case where the input we're comparing with is
+		// ( or has been copied from ) the original string we were initialized with.
+		const std::string *jsonStringAddress = &key.loadedFromCryptomatteJSONStringData->readable();
+		size_t jsonStringAddressCast;
+		memcpy( &jsonStringAddressCast, &jsonStringAddress, 8 );
```

Comment:
I think this can be simpler : https://github.com/GafferHQ/gaffer/pull/6402/commits/743751dc6a4a6de9a5cc869201331a8c8d01adf8